stage1/init/container.go

```diff
@@ -135,7 +135,46 @@ func (c *Container) appToSystemd(ra *schema.RuntimeApp, am *schema.ImageManifest
 		return fmt.Errorf("unable to write environment file: %v", err)
 	}
 
-	execWrap := []string{"/diagexec", common.RelAppRootfsPath(id), workDir, RelEnvFilePath(id)}
+	// Create the Unix user and group
+	var prefix string
+	var username, groupname string
+
+	if app.User == "root" || app.User == "0" {
+		username = "root"
+	} else {
+		username = "u" + app.User
+	}
+	if app.Group == "root" || app.Group == "0" {
+		groupname = "root"
+	} else {
+		// if app.User == app.Group, then systemd-sysusers requires that
+		// username == groupname because it automatically creates a
+		// group for the new user.
+		groupname = "u" + app.Group
+	}
+
+	if err := os.MkdirAll(path.Join(common.Stage1RootfsPath(c.Root), "usr/lib/sysusers.d"), 0755); err != nil {
+		return err
+	}
+
+	sysusersConfig := fmt.Sprintf("%su %s %s \"%s\"\n%sg %s %s\n",
+		prefix, username, app.User, username,
+		prefix, groupname, app.Group)
+
+	err := ioutil.WriteFile(path.Join(common.Stage1RootfsPath(c.Root), "usr/lib/sysusers.d", ServiceUnitName(id)+".conf"),
+		[]byte(sysusersConfig), 0640)
+	if err != nil {
+		return err
+	}
+
+	// - prepare-app bind mounted stage1's diagexec on "/.diagexec" in stage2
+	// - use systemd option "RootDirectory" for chrooting the application,
+	//   so diagexec does not have to chroot. Pass "" as chroot parameter.
+	// - chdir to workDir
+	// - since diagexec runs in the chrooted environment, it does not have
+	//   access to the env file. Pass "" and let EnvionmentFile do the work.
+	execWrap := []string{"/.diagexec", "", workDir, ""}
```

Comment:
We should just own a directory somewhere in the tree for random stuff like this. Ideally it would be `/run` but I don't think this will work b/c the tmpfs hasn't been setup yet. I am ok with something like `/rkt` or something.